projects/Mallard/src/screens/settings-screen.tsx

```diff
@@ -20,28 +19,45 @@ import {
 } from 'src/authentication/AccessContext'
 import DeviceInfo from 'react-native-device-info'
 import {
-    useIsWeatherShown,
     setIsWeatherShown,
-} from 'src/hooks/use-is-weather-shown'
+    setIsUsingProdDevtools,
+} from 'src/helpers/settings/mutators'
+import { useQuery } from 'src/hooks/apollo'
+import gql from 'graphql-tag'
+import ApolloClient from 'apollo-client'
 
-const MiscSettingsList = React.memo(() => {
-    const isShown = useIsWeatherShown()
-    if (isShown == null) return null
-    const onChange = () => setIsWeatherShown(!isShown)
-    const items = [
-        {
-            key: 'isWeatherShown',
-            title: 'Display Weather',
-            data: { onPress: onChange },
-            proxy: <Switch value={isShown} onValueChange={onChange} />,
-        },
-    ]
-    return <List onPress={({ onPress }) => onPress()} data={items} />
-})
+type QueryData = { isWeatherShown: boolean; isUsingProdDevtools: boolean }
+
+const QUERY = gql`
+    {
+        isWeatherShown @client
+        isUsingProdDevtools @client
+    }
+`
+
+const MiscSettingsList = React.memo(
+    (props: { isWeatherShown: boolean; client: ApolloClient<object> }) => {
+        const onChange = () =>
+            setIsWeatherShown(props.client, !props.isWeatherShown)
```

Comment:
Again, having the client as a prop seems a bit odd here. We could either wrap in a custom hook or use a hook to fetch it directly? Maybe I'm missing a use case here though?

Reply:
Simplest could be to make `client` global?

Making custom hooks for each setter would be annoying if you need several of them because you'd need to call each of them separately, whereas with `useApolloClient`, you only need the `client` and that's it.

Reply:
I think I'd be inclined to create a `useWithApollo` hook:

```ts
const useWithApollo = <A extends any[]>(fn: (client: ApolloClient<objects>, ...args: A), ...args: A) => fn(useApolloClient(), ...args);

Reply:
Discussed offline. Becomes a bit complex type and usage wise, we settled on keeping the current mechanism.